tests/acceptance/bootstrap/SearchContext.php

```diff
@@ -220,4 +218,84 @@ public function theSearchResultShouldContainEntriesWithHighlight(
 			);
 		}
 	}
+
+	/**
+	 * @When /^user "([^"]*)" searches for "([^"]*)" inside (folder|space) "([^"]*)"(?: in space "([^"]*)")? using the WebDAV API$/
+	 *
+	 * @param string $user
+	 * @param string $pattern
+	 * @param string $scope
+	 * @param string|null $scopeName
+	 * @param string|null $spaceName
+	 *
+	 * @return void
+	 */
+	public function userSearchesInsideFolderOrSpaceUsingWebDavAPI(
+		string    $user,
+		string    $pattern,
+		string    $scope,
+		string    $scopeName = null,
+		string    $spaceName = null
+	): void {
+		// NOTE: because indexing of newly uploaded files or directories with ocis is decoupled and occurs asynchronously
+		// short wait is necessary before searching
+		sleep(5);
+		$user = $this->featureContext->getActualUsername($user);
+		$baseUrl = $this->featureContext->getBaseUrl();
+		$password = $this->featureContext->getPasswordForUser($user);
+		if (str_contains($pattern, '$')) {
+			$date = explode("$", $pattern);
+			switch ($date[1]) {
+				case "today":
+					$pattern = $date[0] . date('Y-m-d', strtotime('today'));
+					break;
+				case "yesterday":
+					$pattern = $date[0] . date('Y-m-d', strtotime('yesterday'));
+					break;
+				default:
+					throw new Exception("Cannot convert the date");
+			}
+		}
+
+		$body = "<?xml version='1.0' encoding='utf-8' ?>\n" .
+			"<oc:search-files xmlns:a='DAV:' xmlns:oc='http://owncloud.org/ns'>\n" .
+			"  <oc:search>\n";
+
+		if ($scope === "folder") {
+			$resourceID = $this->featureContext->spacesContext->getResourceId($user, $spaceName ?? "Personal", $scopeName);
+			$pattern .= " scope:$resourceID";
+		} elseif ($scope === "space") {
+			// Get the space ID based on the space name
+			$spaceId = $this->featureContext->spacesContext->getSpaceIdByName($user, $scopeName);
+			$pattern .= " scope:$spaceId";
+		}
+
+		$body .= "<oc:pattern>$pattern</oc:pattern>\n" .
+				"  </oc:search>\n" .
+				"</oc:search-files>";
+
+		$davPathVersionToUse = $this->featureContext->getDavPathVersion();
+		$davPath = WebDavHelper::getDavPath($user, $davPathVersionToUse, 'files');
+
+		// If using the "new" DAV path, strip the username part from the path
+		if ($davPathVersionToUse == WebDavHelper::DAV_VERSION_NEW) {
+			// Removes the last component('username' in this case) from the WebDAV path by going up one level in the directory structure.
+			// e.g. remote.php/dav/files/Alice ==> remote.php/dav/files/
+			$davPath = \dirname($davPath, 1);
+		}
+
+		$fullUrl = WebDavHelper::sanitizeUrl("$baseUrl/$davPath");
+		$response = HttpRequestHelper::sendRequest(
+			$fullUrl,
+			$this->featureContext->getStepLineRef(),
+			'REPORT',
+			$user,
+			$password,
+			null,
+			$body
+		);
+
+		// Store the response for further steps to use
```

Comment:
These steps are used everywhere in the codebase. Comment only when necessary otherwise the test file would be filled with comment
```suggestion
		
```